stephenPrintup/README.md

```diff
@@ -0,0 +1,276 @@
+### Homework #1
+(Due Monday 7/20)
+
+You need to turn in this homework by 
+
+[X]cloning this repo, 
+[X]making a new branch with your solutions, 
+[X]pushing to github, and 
+[X]issuing a pull request.
+
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_30% of total time_)
+Predict the output of each of the following expressions, and explain the rules which determine that answer.  (_Hint: The rules for operator `==` are complicated, but do your best!_)
+
+If the output depends on the value or type of variable _x_, identify the conditions (what _x_ is) when the expression will output true (or false, if that's simpler).  Assume the cases are independent, and _x_ is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+
+prediction 	11							<---wrong
+result 		true
+
+**b)** `"1" === 1`
+
+prediction	false
+result      false
+
+**c)** `x == 'x'`
+
+prediction true								<----wrong
+result     undefined
+
+**d)** `x == (x+'')`
+
+prediction undefined
+result     undefined
+
+**e)** `'' == ' '`
+
+prediction  true							<----wrong
+result      false
+
+**f)** `x = true`
+
+prediction  true
+result      true
+
+**g)** `var x; x == 'undefined'`
+
+prediction  undefined							<----wrong
+result      false
+
+**h)** `'9'<'10'`
+
+prediction true
+result     false 				            	<---reveiw
+
+**i)** `typeof x + 1 === "number"`
+
+prediction false, because "number" is a string
+result     false
+
+**j)** `typeof x % 2 === "number"`
+
+prediction false, because "number" is a string
+result     false
+
+**k)** `typeof (x % 2) === "number"`
+
+prediction false, because "number" is a string
+result    true                      				<----review
+
+**l)** `x++ == ++x`
+
+prediction false, x++ will add 1 and ++x will turn x into number
+result     false
+
```

Comment:
Not quite right. False is the correct answer, but depending on what the value of x is, it may already be a number. The situation that occurs here is that if, say, x = 1. x++ returns 1 (while also adding 1 to the value of x) and is then compared to ++x. ++x, however, returns 3, because the left side (x++) caused x to = 2, and the right side (++x) added 1 to x before turning its value (which at that point becomes 3).  If x is a string, then x++ or ++x returns NaN, which is never == to itself.